tests/tasks/test_tasks.py

```diff
@@ -43,8 +43,8 @@ def wait_for_task_completion():
         while task_state in WAIT_STATUSES:
             if elapsed_seconds == TIME_LIMIT:
                 raise requests.Timeout
-            sleep(0.5)
-            elapsed_seconds += 0.5
+            sleep(1)
```

Comment:
[Here](https://stackoverflow.com/questions/492519/timeout-on-a-function-call) are some other solutions to handle timeouts. Perhaps it might be better to define a context manager or decorator, as it's more portable/reusable. However, this is just a FYI, no need to do this in this PR (or at all).